test/test_active.py

```diff
@@ -0,0 +1,279 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+import itertools
+import logging
+import math
+import unittest
+from test.multiprocess_test_case import MultiProcessTestCase, get_random_test_tensor
+
+import crypten
+import crypten.communicator as comm
+import torch
+import torch.nn.functional as F
+from crypten.common.rng import generate_kbit_random_tensor, generate_random_ring_element
+from crypten.common.tensor_types import is_float_tensor
+from crypten.common.util import pool2d_reshape
+from crypten.mpc import MPCTensor, ptype as Ptype
+from crypten.mpc.primitives import ArithmeticSharedTensor, BinarySharedTensor
+
+
+class TestActive(object):
+    """
+    This class tests all functions of MPCTensor.
+    """
+
+    def _get_random_test_tensor(self, *args, **kwargs):
+        return get_random_test_tensor(device=self.device, *args, **kwargs)
+
+    def _check(self, encrypted_tensor, reference, msg, dst=None, tolerance=None):
+
+        if tolerance is None:
+            tolerance = getattr(self, "default_tolerance", 0.05)
+        tensor = encrypted_tensor.get_plain_text(dst=dst)
+        if dst is not None and dst != self.rank:
+            self.assertIsNone(tensor)
+        else:
+            # Check sizes match
+            self.assertTrue(tensor.size() == reference.size(), msg)
+
+            self.assertTrue(is_float_tensor(reference), "reference must be a float")
+
+            if tensor.device != reference.device:
+                tensor = tensor.cpu()
+                reference = reference.cpu()
+
+            diff = (tensor - reference).abs_()
+            norm_diff = diff.div(tensor.abs() + reference.abs()).abs_()
+            test_passed = norm_diff.le(tolerance) + diff.le(tolerance * 0.1)
+            test_passed = test_passed.gt(0).all().item() == 1
+            if not test_passed:
+                logging.info(msg)
+                logging.info("Result %s" % tensor)
+                logging.info("Reference %s" % reference)
+                logging.info("Result - Reference = %s" % (tensor - reference))
+            self.assertTrue(test_passed, msg=msg)
+            self.assertTrue(hasattr(encrypted_tensor, "_mac"), "_mac attribute failed to propagate")
+
+        if dst is None or dst == self.rank:
+            tensor = encrypted_tensor._tensor.reveal(dst=dst)
+            alpha = MPCTensor.alpha.reveal(dst=dst)
```

Comment:
From CircleCI:

```
~ Test test.test_active.TestTTP.test_encrypt_decrypt failed ~
===========
Exceptions:
===========
** Process 0 ** :
Traceback (most recent call last):
  File "/home/circleci/CrypTen/test/multiprocess_test_case.py", line 199, in _run
    getattr(self, test_name)()
  File "/home/circleci/CrypTen/test/test_active.py", line 98, in test_encrypt_decrypt
    self._check(encrypted_tensor, reference, "en/decryption failed")
  File "/home/circleci/CrypTen/test/test_active.py", line 65, in _check
    alpha = MPCTensor.alpha.reveal(dst=dst)
AttributeError: 'NoneType' object has no attribute 'reveal'
```

Looks like in `TestTTP.test_encrypt_decrypt`, we are getting `MPCTensor.alpha = None`.

`MPCTensor.alpha` should be set on line 97, but it is not being set. It appears this is because `TestTTP.setUp` does not set `MPCTensor.ACTIVE_SECURITY = True` as we do with `TestTFP.setUp` on line 250.